modules/ipc/include/hephaestus/ipc/zenoh/action_server/action_server.h

```diff
@@ -163,11 +166,21 @@ ActionServer<RequestT, StatusT, ReplyT>::ActionServer(SessionPtr session, TopicC
                  .status = serdes::getSerializedTypeInfo<StatusT>() })
   , type_info_service_(createTypeInfoService(
         session_, topic_config_, [this](const std::string&) { return this->type_info_.toJson(); }))
-  , request_consumer_([this](const Request<RequestT>& request) { return execute(request); }, std::nullopt) {
+  , request_consumer_([this](const Request<RequestT>& request) { return execute(request); }, std::nullopt)
+  , stop_service_(std::make_unique<Service<std::string, std::string>>(
+        session_, internal::getStopServiceTopic(topic_config_),
+        [this](const std::string&) {
+          this->stop_requested_ = true;
```

Comment:
we need to be carefull, here if we call stop and the service is not running, but then we start running the stop_requested_ would be true and we will never execute the request